llvm/clang warnings: implicit conversion

These 4 warnings occur while compiling some of the source files:
```
In file included from ./ngtcp2/lib/ngtcp2_conv.c:32:
./ngtcp2/lib/ngtcp2_net.h:112:14: warning: implicit conversion changes signedness: 'int' to 'uint32_t' (aka 'unsigned int') [-Wsign-conversion]
  res = *p++ << 24;
      ~ ~~~~~^~~~~
./ngtcp2/lib/ngtcp2_net.h:113:15: warning: implicit conversion changes signedness: 'int' to 'unsigned int' [-Wsign-conversion]
  res += *p++ << 16;
      ~~ ~~~~~^~~~~
./ngtcp2/lib/ngtcp2_net.h:114:15: warning: implicit conversion changes signedness: 'int' to 'unsigned int' [-Wsign-conversion]
  res += *p++ << 8;                                      
      ~~ ~~~~~^~~~                                    
./ngtcp2/lib/ngtcp2_net.h:122:14: warning: implicit conversion loses integer precision: 'int' to 'uint16_t' (aka 'unsigned short') [-Wimplicit-int-conversion]
  res = *p++ << 8;                                    
      ~ ~~~~~^~~~                                     
4 warnings generated.                                 
```

Compiler: llvm/clang 15.0.0 cross-compiling for Windows x64.

--- a/lib/ngtcp2_net.h
+++ b/lib/ngtcp2_net.h
@@ -109,9 +109,9 @@ STIN uint16_t ngtcp2_htons(uint16_t hostshort) {
 STIN uint32_t ngtcp2_ntohl(uint32_t netlong) {
   uint32_t res;
   unsigned char *p = (unsigned char *)&netlong;
-  res = *p++ << 24;
-  res += *p++ << 16;
-  res += *p++ << 8;
+  res = (uint32_t)(*p++ << 24);
+  res += (uint32_t)(*p++ << 16);
+  res += (uint32_t)(*p++ << 8);
   res += *p;
   return res;
 }
@@ -119,7 +119,7 @@ STIN uint32_t ngtcp2_ntohl(uint32_t netlong) {
 STIN uint16_t ngtcp2_ntohs(uint16_t netshort) {
   uint16_t res;
   unsigned char *p = (unsigned char *)&netshort;
-  res = *p++ << 8;
+  res = (uint16_t)(*p++ << 8);
   res += *p;
   return res;
 }

